VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +123,90 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    private class CommentsLoader {
+
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private ThreadComments unendorsedComments;
+        private int nextPage = 1;
+        private boolean isQuestionTypeThread = false;
+        private boolean isEndorsedFetched = false;
```

Comment:
This flag is unnecessary. We can simply (re)load the endorsed comments based on `nextPage` having a value of 1, which would cover the case where the loader state is reset as well.